jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/codeAction/proposal/RemoveMultipleAnnotations.java

```diff
@@ -0,0 +1,82 @@
+/*******************************************************************************
+* Copyright (c) 2022 IBM Corporation and others.
+*
+* This program and the accompanying materials are made available under the
+* terms of the Eclipse Public License v. 2.0 which is available at
+* http://www.eclipse.org/legal/epl-2.0.
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Contributors:
+*     IBM Corporation, Adit Rada - initial API and implementation
+*******************************************************************************/
+package org.eclipse.lsp4jakarta.jdt.codeAction.proposal;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.jdt.core.dom.ASTNode;
+import org.eclipse.jdt.core.dom.IBinding;
+import org.eclipse.lsp4j.CodeAction;
+import org.eclipse.lsp4j.Diagnostic;
+import org.eclipse.lsp4jakarta.jdt.codeAction.JavaCodeActionContext;
+import org.eclipse.lsp4jakarta.jdt.codeAction.proposal.quickfix.RemoveAnnotationConflictQuickFix;
+
+import com.google.gson.JsonArray;
+
+/**
+ * This class is used to provide options to remove multiple annotations
+ * at the same time. For example, "Remove @A, @B", "Remove @C, @D, @E".
+ * 
+ * @author Adit Rada
+ *
+ */
+public abstract class RemoveMultipleAnnotations extends RemoveAnnotationConflictQuickFix {
+    
+    public RemoveMultipleAnnotations() {
+        // annotation list to be derived from the diagnostic passed to
+        // `getCodeActions()`
+        super();
+    }
+    
+    @Override
+    public List<? extends CodeAction> getCodeActions(JavaCodeActionContext context, Diagnostic diagnostic,
+            IProgressMonitor monitor) throws CoreException {
+        ASTNode node = context.getCoveredNode();
+        IBinding parentType = getBinding(node);
+
+        JsonArray diagnosticData = (JsonArray) diagnostic.getData();
+
+        List<String> annotations = IntStream.range(0, diagnosticData.size())
+                .mapToObj(idx -> diagnosticData.get(idx).getAsString()).collect(Collectors.toList());
+
+        if (parentType != null) {
+            List<CodeAction> codeActions = new ArrayList<>();
+            
+            List<List<String>> annotationsListsToRemove = getMultipleRemoveAnnotations(annotations);
+            for (List<String> annotationList : annotationsListsToRemove) {
+                String[] annotaions = annotationList.toArray(new String[annotationList.size()]);
```

Comment:
`annotaions` sorry to be so picky but you may want to fix this typo if you have time.

Reply:
Thank you for catching that. Done